test/basictests.jl

```diff
@@ -39,6 +39,7 @@ function test_interface(alg, prob1, prob2)
     sol = solve(cache; cache_kwargs...)
     @test A2 * sol.u ≈ b1
 
+    cache = LinearSolve.set_A(cache, deepcopy(A2))
```

Comment:
this is no longer testing that the no-aliasing works.

Reply:
I added that to make the last test work for `cholesky!`. Does that indicate that the no-aliasing behavior is not correctly defined for this algorithm? Where?

Reply:
Seems like it. `default_alias_A(::Any, ::Any, ::Any) = false` should make it internally deepcopy. This this is a separate topic so it shouldn't be in the same PR.

Reply:
Yeah, I saw that and thought that it should do the deep copy internally, but the failing test tells me that it isn't *shrug*. I reverted back to not deep-copying here and re-commented the failing test.